### PR TITLE
Create log analytics workspace to be used by key vault diagnostics

### PR DESCRIPTION
**Pull request by mrzta**
**KeyVault logs to Log Analytics Workspace**

- This PR creates a new Log Analytics workspace
- Sets the diagnostics settings in the KeyVault and enables the diagnostics 
- Enable diagnostics by setting the value on enableDiagnostics = true defaults to false

### CODE REVIEW

File: azure/templates/log-analytics.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "logAnalyticsWorkspaceName": {
+      "type": "string",
+      "metadata": {
+        "description": "Specify the name of the workspace."
```
Comment (nickdevdfe) [inline azure/templates/log-analytics.json]: We we add an example, EG
"description": "Specify the name of the workspace. eg: faltrn-kv-law"

Comment (mrzta) [reply azure/templates/log-analytics.json]: This is an input parameter unfortunately "logAnalyticsWorkspaceName=${RESOURCE_NAME_PREFIX}-${SERVICE_SHORT}-${CONFIG_SHORT}-log" which is what will need to be set if the enable diagnostics is set to true

**Comment by nickdevdfe** [comment]
# Errors deploying more than one key vault

```
arm-deployment: set-resource-group-name set-storage-account-name set-azure-template-tag set-azure-account set-azure-resource-group-tags set-key-vault-names ## deploy container/kv to store terraform state for each environment
	az deployment sub create --name "resourcedeploy-tsc-$(shell date +%Y%m%d%H%M%S)" \
		-l "UK South" --template-uri "https://raw.githubusercontent.com/DFE-Digital/tra-shared-services/9b29f223e5b0019a2bc94cb1653e2b74a9934742/azure/resourcedeploy.json" \
		--parameters "resourceGroupName=${RESOURCE_GROUP_NAME}" 'tags=${RG_TAGS}' \
			"tfStorageAccountName=${STORAGE_ACCOUNT_NAME}" "tfStorageContainerName=${SERVICE_SHORT}-tfstate" \
			keyVaultName="${KEY_VAULT_APPLICATION_NAME}", "${KEY_VAULT_INFRASTRUCTURE_NAME}")' \
			"location=uksouth" "logAnalyticsWorkspaceName=s189d01-rtt-rv-kv-log" "enableDiagnostics=true" \
			"enableKVPurgeProtection=${KEY_VAULT_PURGE_PROTECTION}" ${WHAT_IF}
```
Will generate an error because more than one key vault is specified. If set to single KV it works fine.

Error code is:
```
MultipleErrorsOccurred - Multiple error occurred: BadRequest,BadRequest. Please see details.
BadRequest - Deployment template validation failed: 'The value for the template parameter 'logAnalyticsWorkspaceName' at line '20' and column '34' is not provided. Please see https://aka.ms/arm-create-parameter-file for usage details.'.
BadRequest - Deployment template validation failed: 'The value for the template parameter 'logAnalyticsWorkspaceName' at line '20' and column '34' is not provided. Please see https://aka.ms/arm-create-parameter-file for usage details.'.
make: *** [Makefile:257: arm-deployment] Error 1
```

@saliceti @mrzta

**Review by nickdevdfe — APPROVED** [review]
Tested, reviewed and approved.

**Pull request merged**